### ISSUE TITLE
Accessing StateMachine.PermittedTriggers throws System.ArgumentException

### ISSUE THREAD

**Issue by rose-a** [dotnet-state-machine/stateless#275]
Accessing the `StateMachine.PermittedTriggers` property throws a `System.ArgumentException` with the message `An argument of type System.String is required in position 0.`

StackTrace:
```
   bei Stateless.ParameterConversion.Unpack(Object[] args, Type argType, Int32 index)
   bei Stateless.ParameterConversion.Unpack[TArg](Object[] args, Int32 index)
   bei Stateless.StateMachine`2.TransitionGuard.<>c__DisplayClass4_0`1.<ToPackedGuard>b__0(Object[] args)
   bei Stateless.StateMachine`2.TransitionGuard.<>c__DisplayClass17_0.<UnmetGuardConditions>b__0(GuardCondition c)
   bei System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   bei System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   bei System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   bei Stateless.StateMachine`2.TransitionGuard.UnmetGuardConditions(Object[] args)
   bei Stateless.StateMachine`2.StateRepresentation.<>c__DisplayClass69_0.<GetPermittedTriggers>b__2(TriggerBehaviour a)
   bei System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   bei Stateless.StateMachine`2.StateRepresentation.<>c__DisplayClass69_0.<GetPermittedTriggers>b__0(KeyValuePair`2 t)
   bei System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   bei System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   bei System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   bei Stateless.StateMachine`2.StateRepresentation.GetPermittedTriggers(Object[] args)
   bei Stateless.StateMachine`2.get_PermittedTriggers()
```

**Comment by kpi6288** [comment]
I can verify this for the project TelephoneCallExample. Appearently the last version that does not show this exception is the commit dated Mar 24, 2018.

**Comment by smartcaveman** [comment]
@rose-a @kpi6288 @matt-w-cotton @HenningNT - I submitted a PR with a patch and related tests that demonstrate the conditions under which the problem happens.  There are still some issues related to calling `StateMachine.GetPermittedTriggers(object[])`.  The intent of what that method **should do** was not very clear to me, but these changes will fix cases where `StateMachine.ToString()`, `StateMachine.PermittedTriggers` and `StateMachine.CanFire(TTrigger)` all currently throw unexpected exceptions.

**Comment by HenningNT** [comment]
Will be fixed in 5.1.0.
